### ISSUE TITLE
 Can we get mat_picmip -10 back? (Ultra high textures)

### ISSUE THREAD

**Issue by Freeman-51** [ValveSoftware/Source-1-Games#699]
Can we get mat_picmip -10 back? (Ultra high textures)
It's been nearly 3 years now since VALVe locked mat_picmip to -1 to 4. mat_picmip -10 enabled the highest texture resolutions to be used in the game, adding some fine details to some textures (like the clothing of the classes) and removing some hideous noticeable compression artifacts on others (like everything).

mat_picmip was locked, according to the wiki, because of "significant performance and instability issues". Which is a pretty silly reason on its alone. People should know messing in the console is done at their own risk! Besides that, hardware has improved dramatically in the past 3 years. Surely it's not unreasonable to expect a large sum of people are able to use this without 'performance issues'.

Quoting Metal Geo on the steam forum.
I'm pretty much agree with him.

http://forums.steampowered.com/forums/showthread.php?t=3109514

**Comment by agrastiOs** [comment]
Yeah, could it be unlocked? @kisak-valve, it should get the "Team Fortress 2" label.

**Comment by shoober420** [comment]
I needs my high quality -10 picmips!

**Comment by memmam** [comment]
Just tagging a message on here hoping it'll get noticed by whoever has been actively working on TF2. mat_picmip has been locked to a range of -1 to 4 for a very long time and given the current hardware landscape I don't think that's necessary anymore.

**Comment by memmam** [comment]
Just tagging a message on here hoping it'll get noticed by whoever has been actively working on TF2 (seems like @kisak-valve?). mat_picmip has been locked to a range of -1 to 4 for a very long time and given the current hardware landscape I don't think that's necessary anymore.

**Comment by kisak-valve** [comment]
I'm not the Team Fortress 2 developer you're looking for, but "Updated mat_picmip to allow values below -1" is one of the changelog notes in the [2022-07-07](https://www.teamfortress.com/post.php?id=140004) Team Fortress 2 update.

**Comment by stevethepocket** [comment]
@chipsnapper Do you know of any textures in any of those games that would actually be higher res if you did that? Otherwise it would be a completely useless change.

**Comment by chipsnapper** [comment]
@stevethepocket While this doesn't have any noticeable effects on world textures, it can affect the texture quality of small prop models. I can't give you an example of a model from Half-Life 2 right now (because the max value is still -1) but here's an example from TF2:
![image](https://user-images.githubusercontent.com/3721368/178077363-c13e87e6-8db3-4e31-9fab-27f38b4eb46b.png)
Similar situations exist in HL2 on props like soda cans, trash, etc.

**Comment by Ashetf2** [comment]
@kisak-valve Can you close this issue? It seems that OP hasn't been active in years

**Comment by agrastiOs** [comment]
@Ashetf2 It still works for other Source 1 games.

**Comment by CptFoxer** [comment]
Agreed with chipsnapper, It's already been enabled in TF2, it should be enabled in all other Source 1 games (unsure if this also includes the SDK Base 2007, and 2013 branches, but will assume they also count)

**Comment by chipsnapper** [comment]
SDK2013 reflects the TF2 codebase, it should get this change whenever it gets updated next. SDK2007 is no longer updated however.

**Comment by Wadmodder** [comment]
Since the release of macOS Catalina which removed support for 32-bit Carbon programs, Valve should reimplement mat_picmip -10 in all Source 1 games, since they lacks the staff necessary to update their macOS ports to 64-bit. The limit of mat_picmip -1 was previously introduced back in 2010, and should be removed due to the release of macOS Catalina and Valve's lack of resources regarding the Apple Silicon processor architecture.

**Comment by Wadmodder** [comment]
Hey @kisak-valve, Since Valve recently removed the macOS binaries for their Source engine games, why not reimplement mat_picmip -10 for both Windows & Linux. Or maybe the mat_picmip -10 update for all the other Source engine games have been delayed indefinitely.

**Comment by AwesomeCoder412412** [comment]
It is in fact back in team fortress 2, not sure about other source games.

**Comment by Wadmodder** [comment]
@kisak-valve Even with the removal of macOS binaries from Valve's titles, I probably think that mat_picmip -10 update for all the other Source engine games has either been delayed indefinitely, or cancelled completely.

**Comment by shoober420** [comment]
for the sake of preservation, and freedom for tweaking the game and its visuals in a fair competitive manner, this needs added back. there is no competitive advantage to have high quality textures, in fact the exact opposite. quake players use high positive picmip values (2+) to make it easier to see players models.

but if one chooses to play with high quality negative -10 picmip for fun and to see the full extent of texture quality, they should be allowed to do so.

there is no cheating involved when playing with higher quality textures, so theres no reason to block their use. its all for fun and to see the full extent of the engines capabilities.

theres no reason to lock this cvar in this manner. please allow us to play with -10 picmip so we can enjoy playing with full HD quality textures without downloading mods.

theres some nights i feel competitive, and want to play with positive picmip 3, and some nights i want to have fun and play with negative -1 picmip.

please allow -10 picmip!

**Comment by Wadmodder** [comment]
mat_picmip -10 update for remaining Source 1 games cancelled indefinitely.

**Comment by Wadmodder** [comment]
@kisak-valve I think the whole mat_picmip -10 update for remaining Source 1 games has been cancelled indefinitely.

**Comment by memmam** [comment]
@Wadmodder just so you know, harassing the person at Valve who runs this repo (and all of us who get emails every time a post is made in this issue, tbf) is really not helping things.

**Comment by shoober420** [comment]
@memmam
saying "harassing" is a stretch. hes a mod afterall, he gets notifications all the time. if you dont want emails and notifications about the thread, unsubscribe

i wish the pic of mips was in the minus 10s

**Comment by memmam** [comment]
@shoober420 repeatedly bumping an issue when you have nothing meaningful to contribute is not reasonable behavior. Posting the same basic comment twice, a passive-aggressive 'guess it's not happening', a month apart, likely with the intent to keep doing so, is harassment.

I agree with wanting to bring mat_picmip -10 back, but the fact of the matter is Valve has no responsibility to any of us, and poking someone over and over when for all we know it's already being planned if not worked on is not endearing anyone to that cause.

**Comment by shoober420** [comment]
@memmam
he didnt tag kisak in the first post. he was just letting him know personally in the second post. if he tagged him twice in each post i could see it being looked at as blowing up his notifications. assuming hes going to repeatedly tag him and spam post is not valid to consider harassment unless he actually does so. hes more than likely just trying to get him to close the issue.

**Comment by gdfsgdfg** [comment]
Shouldn't this thread be locked we got picmip unlocked years ago.

**Comment by memmam** [comment]
@gdfsgdfg Only in TF2. The continued posts are asking for it to be unlocked in the other Source Engine titles covered by this repository.

**Comment by Wadmodder** [comment]
I think all of Valve's GitHub crew has been reduced to just a skeleton crew long after COVID and long after the TF2 bot crisis. In otherwords Valve's Source engine GitHub issue tracker is a scam.

**Comment by memmam** [comment]
@Wadmodder Okay, that's not constructive and pretty firmly cements this as harassment on your part. If you keep it up, I wouldn't be surprised if someone marks your comments as spam or abuse and bans you from Valve's repositories.

**Comment by Wadmodder** [comment]
You know what, we'll probably have to wait until the 20th anniversary of either Half-Life 2 Lost Coast in October 2025, Half-Life 2 Episode One in June 2026 or The Orange Box in October 2027 for mat_picmic -10 to be fully restored for all the other Source 1 engine games.

**Comment by shoober420** [comment]
that would be nice, i really like how they updated Half-Life and all GoldSrc titles. its a valve time issue i imagine. we will see -10 picmips eventually
